verify.rb

```diff
@@ -98,31 +81,73 @@ def process_section(section, validator)
     error("#{img} is not used")
   end
 end
-# rubocop:enable AbcSize,MethodLength
+# rubocop:enable MethodLength
+
+def validate_data(validator, data, file, identifier, subset = nil)
+  val = 2
+  if subset.nil?
+    subset = data if subset.nil?
+    val -= 1
+  end
+
+  validator.validate(data).each do |e|
+    msg = parse_error_msg(e, val, subset)
+    error("#{file}:#{subset.at(e.path.split('/')[val].to_i)[identifier]}"\
+          ": #{e.message}#{msg}")
+  end
+end
+
+def parse_error_msg(error, val, subset)
+  msg = ''
+  if error.message.include? " is already used at '/"
+    err_split = error.message.split('already used at')[1].split('/')
+    return "\nThese listings share the same "\
+          "'#{err_split[val + 1].split('\'')[0]}':"\
+          "\n#{subset.at(err_split[val].to_i).to_yaml}"\
+          "#{subset.at(error.path.split('/')[val].to_i).to_yaml}\n"
+  end
+
+  msg
+end
+# rubocop:enable AbcSize
+
+def validate_schema(parser, schema)
+  parser.parse_file(File.join(__dir__, schema))
+  errors = parser.errors()
+  return unless errors && !errors.empty?
+  errors.each do |e|
+    error(e.message.to_s)
+  end
+end
+
+def validate_alphabetical(set, identifier, set_name)
+  return unless set != (set_sort = set.sort_by { |s| s[identifier].downcase })
+  error("#{set_name} not ordered by #{identifier}. Correct order:" \
```

Comment:
Should use this style

```
str =<<-EOS
Whatever you want to print with #{interpolation} support
EOS
```

Reply:
You’re saying set a variable to that value first? Then send to error(..

Reply:
Yep.